### ISSUE TITLE
using <Button> or <Link> instead of <span> 

### ISSUE THREAD

**Issue by alhansrisuk** [gatsbyjs/gatsby#20504]
## Summary
Was looking at some code for `button-expand-all.js` and am wondering if the `Collapse All` and `Expand All` should be wrapped in a `Button` or `Link` instead of a `span` for accessibility purposes? 

https://github.com/gatsbyjs/gatsby/blob/e0f9b39dc96cdd974693261da7518ba21f29a6e7/www/src/components/sidebar/button-expand-all.js

A `<span>` gives us very little information on what the element is through screen readers.

**Comment by alhansrisuk** [comment]
Confirmed that only the text is wrapped in a `<span>`.
